include/chopper/detail_apply_prefix.hpp

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include <filesystem>
+#include <string_view>
+
+namespace chopper::detail
+{
+
+//!\brief Creates a filename and directory name from `prefix`.
+inline void apply_prefix(std::string_view const & prefix,
+                         std::filesystem::path & filename,
+                         std::filesystem::path & directory)
+{
+    filename += prefix;
+    filename += ".weight";
+    directory += prefix;
+    directory += "_sketches";
```

Comment:
Are you sure this always works as intended?

`+=` does not introduce a directory separator.

What happens on input of `some_path/` vs. `some_path`?
Probably works, but not as expected?

Reply:
> `+=` does not introduce a directory separator.

Yes thats what I want

> What happens on input of `some_path/` vs. `some_path`?

Good point. also `some_path.foo` might result in unexpected behaviour. I'll try to protect against it.